Update release notes process (#6869)
keystonejs/keystone: RELEASE.md

Merge the PR named `Version Packages` into `main` on GitHub once tests are passing.

> In the command line, run `git pull` to latest from `main`.

Reviewer: ```suggestion
In the command line, run `git pull` to fetch latest changes from `main`.
```

Run `yarn fresh` to rebuild `node_modules` and clean-up any build output.

Finally, run `yarn publish-changed` to find packages where the version listed in the `package.json` is ahead of the version published on npm, and publish just those packages.

The above command requires OTP and publish privileges on npm.

> Create git tag relevant to release date, such as `git tag -a "YYYY-MM-DD" -m "YYYY-MM-DD"`

Reviewer: ```suggestion
Create git tag relevant to release date, such as `git tag "$(date +'%Y-%m-%d')"`
```

No message needed?

Then publish all new tags (releases plus release date version) to GitHub with `git push --tags`.

Copy the successful output of the published packages from the above npm release, example output:

```sh
@keystone-next/auth@27.0.0
@keystone-next/fields@11.0.0
@keystone-next/keystone@20.0.0

```

> Convert to a `package.json` style syntax for ease of copying, example conversion:

Reviewer: ```suggestion
Translate to a `package.json` style syntax for ease of copying, example conversion:
```

```json
"@keystone-next/auth": "27.0.0",
"@keystone-next/fields": "11.0.0",
"@keystone-next/keystone": "20.0.0",
```

You can safely remove references to `@keystone-ui` as they are primarily for internal use in the Admin UI.

Use this output in the next step.

## Website release

Create a new branch off `website_live`, name it something like `2021-11-02-release-notes`

Duplicate the last release notes for this release under `/docs/pages/releases/YYYY-MM-DD.mdx`.

Copy over the PR content you've processed from the above section under the `## What's New` heading.

Group items and add headings to call out sections to have the release start to take shape, add in `>` quotes as editors notes for ideas on topics.

Open a PR against `website_live` with this branch to allow others to review when there is an overall story.

Finalise content based on feedback, clean up text, add emojis.

Make sure the `Complete Changelog` section has valid links to the latest release.

Publish release on GitHub.

Add release to release page index under `/docs/pages/releases/index.mdx` with a summarised heading.

> Commit website update and open PR to add to `main`.

Reviewer: ```suggestion
Commit your changes; reviewed, then merged.
```

> Have PR reviewed and merged into `main`.

Reviewer: ```suggestion
```

Go through PR descriptions and determine what should be called out in the release, what should be summarised, or left out completely.

## GitHub branch sync

When we do a release we need to make sure `main` and `website_live` are both in sync with each other.

This can be done by checking out `website_live`.

Then running:

```sh

> EDITOR=subl git rebase --interactive origin/main

Reviewer: ```suggestion
git rebase --interactive origin/main
```

[Setting up your preferred editor for `git` is on the first page of the `git` setup guide.](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)

I think we can omit that.

```

Resolve any conflicts then run:

```sh
git push --force
```

> Branches should now be nicely aligned and show `0 | 0` on the branches page <https://github.com/keystonejs/keystone/branches>.

Reviewer: This won't be `0 | 0`,  it'll be N commits ahead of `main` until merge.
The key emphasis is that it is _not_ behind.  `0 behind`.
